domain/charm/types.go

```diff
@@ -20,6 +20,24 @@ type GetCharmArgs struct {
 	Revision *int
 }
 
+// Charm represents a charm from the perspective of the service. This is the
+// golden source of charm information. If the charm changes at the wire format
+// level, we should be able to map it to this struct.
```

Comment:
We *must* be able to map it this struct.

Reply:
I'll fix in the subsequent PR.